test/test-options-cluster.js

```diff
@@ -0,0 +1,42 @@
+// Copyright IBM Corp. 2015. All Rights Reserved.
+// Node module: strong-supervisor
+// This file is licensed under the Artistic License 2.0.
+// License text available at https://opensource.org/licenses/Artistic-2.0
+
+var CPUS = require('os').cpus().length;
+var options = require('../lib/options');
+var tap = require('tap');
+
+tap.test('cluster option validation', function(t) {
+  t.test('valid values', function(t) {
+    check(['node', 'sl-run', '--cluster=10'], 10);
+    check(['node', 'sl-run', '--cluster', '-10'], -10);
+    check(['node', 'sl-run', '--cluster', 'CPU'], CPUS);
+    check(['node', 'sl-run', '--cluster=cpu'], CPUS);
+    t.end();
+
+    function check(argv, expect) {
+      var o = options.parse(argv);
+      t.equal(o.cluster.size, expect);
```

Comment:
I've been using ugly patterns like `t.equal(o && o.cluster && o.cluster.size, expect);` in places like this so that the exception that gets thrown by `o` or `o.cluster` being `null` doesn't toss the useful assertion information out the window. I'm sure you agree that's not the best way to deal with it. Do you have any other ideas for cases like this?

This isn't pretty either, but _slightly_ less offenseive looking than than the `&&` chaining I usually do..
```js
t.match(o, {cluster: {size: expect}});
```